### ISSUE TITLE
How to generate _one_ report for linting multiple Dockerfiles across different folders ?

### ISSUE THREAD

**Issue by przemolb** [hadolint/hadolint#854]
- [ ] This is a bug report
- [x] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior

As part of our Gitlab pipeline we want to lint all the dockerfiles with `hadolint` and generate the [Code Quality Report](https://docs.gitlab.com/ee/ci/testing/code_quality.html#code-quality-reports) using `-f gitlab_codeclimate` parameter.
All is fine as long as all the Dockerfiles are stored in consistent, predictiable folders i.e.:
```
folder-1/Dockerfile
folder-2/Dockerfile
folder-3/Dockerfile
folder-4/Dockerfile
folder-5/Dockerfile
```
and then we can run
```
hadolint -f gitlab_codeclimate */Dockerfile
```

However in our case the Dockerfiles are scattered across different folders structure i.e.:
```
folder-1/sub-1/Dockerfile
folder-2/sub-2/sub-3/Dockerfile
folder-3/sub-4/sub-5/sub-6/Dockerfile
folder-4/sub-4/sub-5/sub-6/sub-7/sub-8/Dockerfile
folder-5/sub-x/Dockerfile
```

What is the best way to create **_one_** report for all those Dockerfiles above ?

**Comment by m-ildefons** [comment]
Hi @przemolb 
I don't think this is within the scope of Hadolint. But you can try and see if a bit of Shell can do that for you:
```bash
find topfolder -name Dockerfile | xargs hadolint -f gitlab_codeclimate
```
This should yield a single report for all Dockefiles somewhere under the `topfolder` directory.
Hope it helps

**Comment by przemolb** [comment]
Thanks @m-ildefons - I expected `xargs` to run dedicated `hadolint` process for each `Dockerfile` and creating its own file for each instance (which is not acceptable for GitLab as it requires just one file) but it turned out it creates just one.
Learning all the time ;-)

**Comment by przemolb** [comment]
And it turned out that `xargs` run (as I expected) is only with L1 parameter - otherwise it runs one instance.
